shared/libraries/opcuatms/opcuatms_client/src/objects/tms_client_device_impl.cpp

```diff
@@ -405,14 +407,82 @@ void TmsClientDeviceImpl::findAndCreateCustomComponents()
         this->components.push_back(val);
 }
 
-FunctionBlockPtr TmsClientDeviceImpl::onAddFunctionBlock(const StringPtr& /*typeId*/, const PropertyObjectPtr& /*config*/)
+DictPtr<IString, IFunctionBlockType> TmsClientDeviceImpl::onGetAvailableFunctionBlockTypes()
 {
-    throw OpcUaClientCallNotAvailableException();
+    const auto fbFolderNodeId = getNodeId("FB");
+    const auto methodNodeId = clientContext->getReferenceBrowser()->getChildNodeId(fbFolderNodeId, "GetAvailableFunctionBlockTypes");
```

Comment:
Could this just be a read-only array string node?

Reply:
We have discussed this and decided to go with a folder node of available function block types.